src/main/java/io/vertx/core/shareddata/impl/LocalAsyncMapImpl.java

```diff
@@ -148,6 +163,25 @@ private void removeIfExpired(K k) {
     return ctx.succeededFuture(h == result);
   }
 
+  @Override
+  public Future<Boolean> replaceIfPresent(K k, V oldValue, V newValue, long ttl) {
+    ContextInternal ctx = vertx.getOrCreateContext();
+    long timestamp = System.nanoTime();
+    long timerId = vertx.setTimer(ttl, l -> removeIfExpired(k));
+    Holder<V> h = new Holder<>(newValue, timerId, ttl, timestamp);
+    Holder<V> result = map.computeIfPresent(k, (key, holder) -> {
+      if (holder.value.equals(oldValue)) {
+        if (holder.expires()) {
+          vertx.cancelTimer(holder.timerId);
+        }
+        return h;
+      }
+      vertx.cancelTimer(timerId);
```

Comment:
You cannot do this here because the handler may be called several times if concurrent modifications are executed.